spring-cloud-stream/src/main/java/org/springframework/cloud/stream/config/BinderFactoryConfiguration.java

```diff
@@ -117,7 +117,11 @@ public BinderTypeRegistry binderTypeRegistry(ConfigurableApplicationContext conf
 				URL url = resources.nextElement();
 				UrlResource resource = new UrlResource(url);
 				for (BinderType binderType : parseBinderConfigurations(classLoader, resource)) {
-					binderTypes.put(binderType.getDefaultName(), binderType);
+					// Exclude TestSupportBinder when detected as the TestSupport binder is expected to use its own
+					// BinderFactory when used.
+					if (!binderType.getDefaultName().equalsIgnoreCase("test")) {
```

Comment:
But in this case, the `TestSupportBinderAutoConfiguration` is already **disabled** and no more beans from this. The `BinderFactoryConfiguration` is the one that is active and creates the `BinderFactory`; and hence if at all we need to make changes, then that has to be in the `BinderTypeRegistry` created from `BinderFactoryConfiguration`